### PR TITLE
Reduce LSST Focus in TAPE Intro Docs

### PR DESCRIPTION
**Pull request by wilsonbb**
## Change Description

Changes the README and readthedocs intro page to deemphasize TAPE's goals of working with LSST timeseries data. While these are still noted as part of the motivation of the project, as noted in https://github.com/lincc-frameworks/tape/issues/365 the current emphasis may cause users to not see TAPE as useful for other big surveys.

While here also changed the default recommended python version from 3.10 to 3.11 as was requested in https://github.com/lincc-frameworks/tape/issues/372

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,13 +13,11 @@
...
+The **Time series Analysis and Processing Engine** (TAPE) is a framework for distributed time series analysis which enables the user to scale their algorithms to large datasets, created to work towards the goal of making [LSST](https://www.lsst.org/about) time series analysis accessible. It allows for efficient and scalable evaluation of algorithms on time domain data through built-in fitting and analysis methods as well as support for user-provided algorithms. TAPE supports ingestion of multiple time series formats, enabling easy access to both LSST time series objects and data from other astronomical surveys.
 
 In short term we are working on two main goals of the project:
-  - Enable ease of access to TimeSeries objects in LSST
   - Enable efficient and scalable evaluation of algorithms on time-domain data
+  - Enable ease of access to TimeSeries objects in LSST
```
Comment (nevencaplar) [inline README.md:20]: Let us again call it time-domain data here

**Review by dougbrn — APPROVED** [review]
Also agree this looks good, pending Neven's suggestion
